### DISCUSSION TITLE
Query character limit is not correct

### DISCUSSION THREAD

**Discussion by UmbraSicarius** [tweepy/tweepy#1643] [Questions] [answered]
I have Academic API credentials and these should allow me to query the full archive search (search_all_tweets) with a query character limit of 1024 characters. When I run my code, the only error I get is a 400 BadRequest error that tells me that my query is too long (at 734 characters) and that the maximum length allowed is 128 characters. Here is my code for reference:
```
api = tweepy.Client(bearer_token="BEARER", consumer_key="C_KEY", consumer_secret="C_SECRET", access_token="A_TOKEN", access_token_secret="A_T_SECRET", wait_on_rate_limit=True)`


def weekOne():
    tweets_list_one = api.search_all_tweets(query="700 character keyword list", start_time="2020-06-01T00:00:00z", end_time="2020-06-08T00:00:00z")
    while(True):
        try:                                                         
           tweet = tweets_list_one.next()
           try:
               collection1.insert_one({'_id':tweet.id, 'created_at':tweet.created_at, 'text':tweet.full_text, 'retweet_count':tweet.public_metrics.retweet_count, 'likes_count':tweet.public_metrics.like_count, 'coordinates':tweet.geo.coordinates.coordinates, 'place_id':tweet.geo.place_id})
               print(str(tweet.created_at) + ", came in truncated")
           except AttributeError:
               collection1.insert_one({'_id':tweet.id, 'created_at':tweet.created_at, 'text':tweet.text, 'retweet_count':tweet.public_metrics.retweet_count, 'likes_count':tweet.public_metrics.like_count, 'coordinates':tweet.geo.coordinates.coordinates, 'place_id':tweet.geo.place_id})
               print(str(tweet.created_at) + ", came in not truncated")
        except StopIteration:
            print("Process 1 is done")
            break
        except Exception as e:
            traceback.print_exc()
            time.sleep(60 * 15)
            continue
```

The gist is that tweets are gathered via the client call, then they are added to a mongoDB database. Everything seems to work except for the character limit. Any help would be appreciated

**Comment by Harmon758** [comment]
Can you provide the full traceback?

Unfortunately, I can't test this myself, as I don't have access to this endpoint.  CC @sparack

**Reply by UmbraSicarius** [reply]
Sorry, I don't have the traceback anymore, but the error code was 400, and the message was that my query was too long (700 something characters) because the max allowed was 128. The max for academic api should be 1024.

**Comment by Harmon758** [comment, marked as answer]
I have access to the endpoint now, and from some testing, it seems like each term has a maximum allowed length of 128 characters.
The entire query, composed of multiple terms, can be up to 1024 characters.
You likely have a single term longer than 128 characters. If that's not the case, you'll need to provide the query you're using so I can test it and attempt to reproduce your issue.
